### ISSUE TITLE
"early_stopping_metric" doesn't seem to work

### ISSUE THREAD

**Issue by Lysimachos** [ThilinaRajapakse/simpletransformers#339]
I am trying multi_label and multi_class classification with early stopping.

I have insert the sklearn.metrics.f1(average = "micro) function into **kwargs with using this code:

               ` kwargs = {"f1_micro": partial(sklearn.metrics.f1_score, average = "micro")}`

The f1_micro score is being calculated on the eval_df and it gets printed while training. But the argument :

"early_stopping_metric" = "f1_micro" doesn't work. It seems that it always calculates the loss function on eval_df for the early stopping functionality.
It is my bad or there is indeed a bug?

**Comment by ThilinaRajapakse** [comment]
Just making sure, you are setting `{"early_stopping_metric": "f1_micro"}` in the model args itself, right? If it's not working with that, it's likely a bug.

**Comment by Lysimachos** [comment]
Greeting Thilina,

Thank you for your reply. Yes with in the model args I set :"early_stopping_metric": "f1_micro" along with ""evaluate_during_training":True. But I don't insert it as a dictionary with in the dictionary of model_args

**Comment by ThilinaRajapakse** [comment]
Looks like you are doing it properly. I'll look into this and see what's going on.

**Comment by mmmjung** [comment]
Any updates on this issue?

**Comment by mmmjung** [comment]
I think that at least one problem with early stopping is in this line:
https://github.com/ThilinaRajapakse/simpletransformers/blob/master/simpletransformers/classification/classification_model.py#L486

It seems it should be `>` instead of `<`

**Comment by djstrong** [comment]
If `early_stopping_delta` is positive then for minimizing metric it should be: 
```
best_eval_metric - results[args["early_stopping_metric"]]
                                > args["early_stopping_delta"]
```

**Comment by mmmjung** [comment]
False alarm...
I had the default `"early_stopping_metric_minimize": True` for accuracy.

**Comment by ThilinaRajapakse** [comment]
I didn't have time to investigate this yet, unfortunately. @mmmjung Is early stopping working as it should once you made the correction?

**Comment by djstrong** [comment]
Yes. The problem is when `early_stopping_delta` is not a zero.

**Comment by krannnn** [comment]
> I am trying multi_label and multi_class classification with early stopping.
> 
> I have insert the sklearn.metrics.f1(average = "micro) function into **kwargs with using this code:
> 
> ```
>            ` kwargs = {"f1_micro": partial(sklearn.metrics.f1_score, average = "micro")}`
> ```
> 
> The f1_micro score is being calculated on the eval_df and it gets printed while training. But the argument :
> 
> "early_stopping_metric" = "f1_micro" doesn't work. It seems that it always calculates the loss function on eval_df for the early stopping functionality.
> It is my bad or there is indeed a bug?

@Lysimachos can you please share the code that you used, or point me in the direction of where to add this ` kwargs = {"f1_micro": partial(sklearn.metrics.f1_score, average = "micro")}` ?

Thanks!

**Comment by Lysimachos** [comment]
Hi @krannnn

A way to insert this is:

```
kwargs = {"f1_micro": partial(sklearn.metrics.f1_score, average = "micro")}
self._model.train_model(train_df, val_df,**kwargs) 
```
Where the train_df and val_df are data frames for train and validation datasets that are going to be used for the training. Ofcourse you need to set up the global arguments accordingly. 

Last time I spend some effort on this (around 7 months ago) I did some work on the code concerning training process as well. I don't know If this have been fixed now. I will give it a look when I find some time.

**Comment by MrRace** [comment]
@Lysimachos  So, In the new version it can work as your expectation when set "early_stopping_metric" = "f1_micro"?
